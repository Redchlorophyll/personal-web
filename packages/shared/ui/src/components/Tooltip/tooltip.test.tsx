import React from 'react';

import '@testing-library/jest-dom';
import { Tooltip } from './index';

import {
  screen,
  renderWithProviders,
  userEvent,
} from 'testing-library-react-custom';

describe('shared - ui - tooltip', () => {
  test('should render properly', () => {
    const { baseElement } = renderWithProviders(
      <Tooltip tooltipContent={'this is tooltip'}>Click me Please!</Tooltip>
    );

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
  test('it should render properly', () => {
    renderWithProviders(
      <Tooltip tooltipContent={'this is tooltip'}>Click me Please!</Tooltip>
    );
    const target = screen.getByText('Click me Please!');

    expect(target).toBeInTheDocument();
  });

  test('it should show tooltip content when being hovered', async () => {
    const user = userEvent.setup();
    renderWithProviders(
      <Tooltip tooltipContent={'this is tooltip'}>Click me Please!</Tooltip>
    );
    const target = screen.getByText('Click me Please!');
    await user.hover(target);
    const tooltipContent = screen.getByText('this is tooltip');

    expect(tooltipContent).toBeInTheDocument();
  });
});
