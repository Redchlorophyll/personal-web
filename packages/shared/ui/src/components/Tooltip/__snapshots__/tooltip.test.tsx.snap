// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared - ui - tooltip should render properly 1`] = `
<body>
  <div>
    <div
      class="group relative w-fit"
    >
      <div>
        Click me Please!
      </div>
      <div
        class="absolute invisible opacity-0 group-hover:visible group-hover:opacity-100 transition-all ease-in-out duration-300"
        style="transform: translateX(0px);"
      >
        <div
          class="bg-black-100 mt-2 py-1 pl-2 pr-4 shadow-[0_0_4px_rgba(29,171,221,0.2)] w-fit whitespace-nowrap rounded-sm dark:text-black-900 after:absolute after:left-[1px] after:top-1 after:border-b-4 after:border-b-black-500 after:w-0 after:h-0 after:border-l-4 after:border-r-4 after:border-l-black-100/0 after:border-r-black-100/0"
        >
          this is tooltip
        </div>
      </div>
    </div>
  </div>
</body>
`;
